### DISCUSSION TITLE
FLOWS: Trigger > Read Data > Trigger Flow > Create Data

### DISCUSSION THREAD

**Discussion by EnzoBitMatrix** [directus/directus#17604] [Archive]
Hi team,

I am hoping someone can help with this one.

I have an event hook that is triggered on an item create.  This works without a problem
I then read data from another collection based on the output from the event hook - this works without a problem and the payload looks like this:

`[
  {
    "status": "published",
    "station_id": "CAPT",
    "station_name": "Captain",
    "station_image": "af376e6e-8ad7-4fe4-88ea-686a83b0146e",
    "station_simulation": [
      1,
      2,
      3,
      4
    ]
  },
  {
    "status": "published",
    "station_id": "COMMS",
    "station_name": "Communications",
    "station_image": "4fab4703-d65f-43f3-b90c-bacdf92d5947",
    "station_simulation": [
      8,
      9,
      10
    ]
  }
]`

Now I start to have issues.  The next step is a trigger flow based on the read data.  I am trying to get the trigger flow to then pass the details to the create data flow, but I only want to pass the "station_id" through to the create.

At the moment, but trigger flow looks like this:

`{
    "data": "{{ $last }}"
}`

This then passes this:
`{
  "payload": {
    "data": "[{\"station_id\":\"CAPT\",\"status\":\"published\",\"station_name\":\"Captain\",\"station_image\":\"af376e6e-8ad7-4fe4-88ea-686a83b0146e\",\"station_simulation\":[1,2,3,4]},{\"station_id\":\"COMMS\",\"status\":\"published\",\"station_name\":\"Communications\",\"station_image\":\"4fab4703-d65f-43f3-b90c-bacdf92d5947\",\"station_simulation\":[8,9,10]}`

I have tried all different variations for the trigger payload, including removing the 'data' option, trying things like $last.payload.data.station_id, and pretty much all other variations of it.

Based on the above, the outcome I want is for the create data flow to create two records in the collection... one with station_id of CAPT, and one with COMMS.

The create data payload I have is:

`{"station_id": "{{$last.payload.data.station_id}}"}`

Again, I have tried all manner of variations of the $last, or $trigger, or $data variable.

If anyone is able to provide some assistance at all it would be greatly appreciated!  If it would help, I can create a video and upload to YouTube with exactly what I have.

Thank you in advance!

**Comment by EnzoBitMatrix** [comment]
Nevermind, I figured it out.
All I needed to do was pass {{$last}} to the trigger flow to send the array as it was.

**Comment by tat-pro** [comment]
It is not clear to me how to use the previous value , e.g. from a previous payload , e.g. there is data [
  {
    "id": "186490302"
    "sort": null,
    "status": "published",
    "user_name": "sigma888",
    "first_name": "S.Ma",
    "date_created": "2024-01-18T12:20:05.000Z",
    "date_updated": null,
    "user_created": "9e7db4ff-2fc6-41c6-bfff-2f3c0b08d4a5",
    "user_updated": null,
    "language_code": "ru"
  }
]  and how do I use them I try differently, {{$last.user_name}} and nothing comes out, I also try to use the key from the event for example {{key_event.user_name}}. I also try to use a key from an event like {{key_event.user_name}} and nothing comes out , please tell me how to use permutations in streams , thank you

**Reply by domdus** [reply]
@tat-pro looks like your example data is an array.
Did you try using {{$last[0].user_name}} or {{key_event[0].user_name}} ?

**Reply by avmesquita** [reply]
But $last[0} is supposed to get only first record.
